### DISCUSSION TITLE
Radiant heat transfer problem

### DISCUSSION THREAD

**Discussion by Gallasd** [idaholab/moose#21452] [Q&A General] [answered]
Hello,
I have some questions about radiation heat transfer that I need to ask you
Here's my 2D radiation geometry, here's the solid, heat transfer to outer space, and the naming of the edges.Obviously, there is surface radiation heat exchange between ce1 and top1, and there is surface radiation heat exchange between ce2 and top2.
![image](https://user-images.githubusercontent.com/107783970/176460051-49b57301-7ca7-457f-97cc-0a85f1e7a68a.png)
There are two viewfactor calculation programs provided by Moose, one is UnobstructedPlanarViewFactor, the other is ViewFactorRayStudy, UnobstructedPlanarViewFactor is for the calculation of the unobstructed surface, and ViewFactorRayStudy is for the calculation of the surface with obstacles.
The viewfactor calculation method suitable for my problem is ViewFactorRayStudy. I tried to use ViewFactorRayStudy to calculate, but I found that the angle coefficient obtained is like this:
![image](https://user-images.githubusercontent.com/107783970/176460947-d7769ca6-0b22-49a4-940a-b8bb6f722069.png)
Moose recognized my solid model as a cavity, so there are only angular coefficients at ce1 and ce2. But my model is solid and the rays should be emitted outwards instead of inwards, I don't know if you have any suggestions to solve this problem.

**Comment by GiudGiud** [comment, marked as answer]
Hello

We only support view factor calculations inside meshed regions currently. So make sure the domain is meshed both inside and outside. 
We will have support for external-to-the-mesh view factors next year

Guillaume
